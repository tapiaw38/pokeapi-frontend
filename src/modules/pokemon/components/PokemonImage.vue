<template>
  <div class="container-image justify-content-center">
    <img :src="imageUrl" :alt="imageId" />
  </div>
</template>

<script setup>
import { computed } from "vue";
// eslint-disable-next-line
const props = defineProps({
  imageId: {
    type: Number,
    default: 1,
  },
});

const imageUrl = computed(
  () =>
    `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${props.imageId}.svg`
);
</script>


<style lang="scss" scoped>
.container-image {
    height: 8rem;

    img {
        height: 100%;
        max-width: 125px;
        margin-top: 6%;
        z-index: 2;
    }
}
</style>