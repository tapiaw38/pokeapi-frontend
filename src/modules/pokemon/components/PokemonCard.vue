<template>
  <div class="card">
    <p data-testid="pokemon-name" class="card-title">
      {{ props.pokemon?.name }}
    </p>
    <pokemon-image :imageId="props.pokemon?.id"></pokemon-image>
    <div class="container-descrition">
      <div class="row align-items-center justify-content-center">
        <template v-for="types in props.pokemon?.types" :key="types">
          <span data-testid="pokemon-type-name" class="badge-primary">{{
            types.type.name
          }}</span>
        </template>
      </div>
      <div class="col">
        <p class="card-description">
          Weight:
          <span data-testid="pokemon-weight">{{ props.pokemon?.weight }}</span>
        </p>
        <p data-testid="pokemon-description" class="card-description">
          {{ props.pokemon?.description ? props.pokemon.description.description : '' }}
        </p>

        <div v-if="props.showMoves">
          <p class="card-description">Moves</p>
          <div
            class="row align-items-center justify-content-center container-moves"
          >
            <template v-for="move in props.pokemon?.moves" :key="move">
              <span data-testid="pokemon-move-name" class="badge-primary">{{ move.move.name }}</span>
            </template>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup>
// components
import PokemonImage from "./PokemonImage.vue";
// eslint-disable-next-line
const props = defineProps({
  pokemon: {
    type: Object,
    default: () => ({}),
  },
  showMoves: {
    type: Boolean,
    default: false,
  },
});
</script>

<style lang="scss" scoped>
.card {
  width: 100%;
  height: 100%;
  max-width: 320px;
  border-radius: 10px;
  border: 1px solid #ccc;
  box-shadow: 0px 0px 10px #ccc;
}
.card .card-title {
  font-size: 1.2em;
  font-weight: bold;
  margin-bottom: 10px;
  color: $secondary;
  text-transform: capitalize;
}
.card .card-text {
  font-size: 1.2em;
  color: $secondary;
}

.container-descrition {
  padding-bottom: 20px;
  padding-top: 20px;
  background-color: $info;
  border-top-left-radius: 40%;
  border-top-right-radius: 40%;

  .card-description {
    font-size: 0.9em;
    font-weight: bold;
    color: $white;
  }
}

.container-moves {
  flex-wrap: nowrap;
  overflow-x: scroll;
  position: relative;
  justify-content: flex-start;
  padding-left: 5px;

  &::-webkit-scrollbar {
    width: 22px;
  }

  &::-webkit-scrollbar-track {
    background: $gray;
    border-radius: 10px;
  }

  &::-webkit-scrollbar-thumb {
    background-color: $info;
    border-radius: 10px;
    border: 2px solid $gray;
  }

  .badge-primary {
    flex-shrink: 0;
    margin-bottom: 5px;
  }
}
</style>
