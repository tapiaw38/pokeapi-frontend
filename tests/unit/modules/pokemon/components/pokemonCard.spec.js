import { shallowMount } from "@vue/test-utils";
import PokemonCard from "@/modules/pokemon/components/PokemonCard.vue";
import PokemonImage from "@/modules/pokemon/components/PokemonImage.vue";

describe("PokemonCard Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonCard);
  });

  test("must match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("pokemon image child component must exist", () => {
    expect(wrapper.findComponent(PokemonImage).exists()).toBeTruthy();
  });

  test("must receive the type of props defined", () => {
    const { pokemon, showMoves } = wrapper.props();
    expect(typeof pokemon).toBe("object");
    expect(typeof showMoves).toBe("boolean");
  });

  test("must receive the pokemon information", () => {
    const pokemon = {
      id: 1,
      name: "bulbasaur",
      types: [{ type: { name: "grass" } }],
      weight: 69,
      description: { description: "bulbasaur description" },
      moves: [{ move: { name: "tackle" } }],
    };

    const showMoves = true;

    const wrapper = shallowMount(PokemonCard, {
      props: {
        pokemon,
        showMoves,
      },
    });
    expect(wrapper.find('[data-testid="pokemon-name"]').text()).toBe(
      pokemon.name
    );
    expect(wrapper.find('[data-testid="pokemon-weight"]').text()).toBe(
      String(pokemon.weight)
    );
    expect(wrapper.find('[data-testid="pokemon-description"]').text()).toBe(
      pokemon.description.description
    );
    expect(wrapper.find('[data-testid="pokemon-type-name"]').text()).toBe(
      pokemon.types[0].type.name
    );
    expect(wrapper.find('[data-testid="pokemon-move-name"]').text()).toBe(
      pokemon.moves[0].move.name
    );
  });

  test("moves must be false if no show description is false", () => {

    const showMoves = false;

    const wrapper = shallowMount(PokemonCard, {
      props: {
        showMoves,
      },
    });
    expect(wrapper.find('[data-testid="pokemon-move-name"]').exists()).toBeFalsy;
  });
});
