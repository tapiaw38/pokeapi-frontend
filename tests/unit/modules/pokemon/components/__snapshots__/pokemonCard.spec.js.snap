// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard Component must match the snapshot 1`] = `
<div class="card">
  <p data-testid="pokemon-name" class="card-title"></p>
  <pokemon-image-stub imageid="1"></pokemon-image-stub>
  <div class="container-descrition">
    <div class="row align-items-center justify-content-center"></div>
    <div class="col">
      <p class="card-description"> Weight: <span data-testid="pokemon-weight"></span></p>
      <p data-testid="pokemon-description" class="card-description"></p>
      <!--v-if-->
    </div>
  </div>
</div>
`;
